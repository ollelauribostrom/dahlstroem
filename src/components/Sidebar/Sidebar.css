.bm-burger-button {
  position: relative;
  width: 30px;
  height: 25px;
}

.bm-burger-bars-hover {
  background: #a90000;
}

.bm-menu {
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
}

.bm-menu-wrap {
  top: 0px;
}

.bm-morph-shape {
  fill: #373a47;
}

.bm-item-list {
  padding: 0.8em;
}

.bm-overlay {
  background: rgba(0, 0, 0, 0.1);
  top: 0;
  left: 0;
}

.sidebar__section {
  margin-bottom: 40px;
}

.sidebar__section__heading {
  margin-bottom: 5px;
  font-size: 16px;
}

.sidebar__link {
  text-decoration: none;
  display: block;
  margin: 15px 0;
  font-size: 14px;
}
