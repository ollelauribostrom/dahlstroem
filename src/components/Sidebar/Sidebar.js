import React from 'react';
import PropTypes from 'prop-types';
import { Link, StaticQuery, graphql } from 'gatsby';
import { push as Menu } from 'react-burger-menu';

const Sidebar = ({ color, menuColor, menuBackground }) => (
  <StaticQuery
    query={graphql`
      query {
        allDatoCmsCategory {
          edges {
            node {
              name
              slug
            }
          }
        }
      }
    `}
    render={data => (
      <Menu
        right
        outerContainerId={'outer-container'}
        styles={{
          bmBurgerBars: {
            background: color.hex,
          },
          bmMenu: {
            background: menuBackground.hex,
          },
        }}
        width={250}
      >
        <div className="sidebar__section">
          <h3 className="sidebar__section__heading">Work</h3>
          {data.allDatoCmsCategory.edges.map(edge => (
            <Link
              to={edge.node.slug}
              style={{ color: menuColor.hex }}
              className="sidebar__link"
            >
              {edge.node.name}
            </Link>
          ))}
        </div>
        <div className="sidebar__section">
          <h3 className="sidebar__section__heading">About</h3>
          <Link
            to="/resume"
            style={{ color: menuColor.hex }}
            className="sidebar__link"
          >
            Resume
          </Link>
          <Link
            to="/contact"
            style={{ color: menuColor.hex }}
            className="sidebar__link"
          >
            Contact
          </Link>
        </div>
      </Menu>
    )}
  />
);

Sidebar.propTypes = {
  color: PropTypes.shape({ hex: PropTypes.string }),
  menuColor: PropTypes.shape({ hex: PropTypes.string }),
  menuBackground: PropTypes.shape({ hex: PropTypes.string }),
};

Sidebar.defaultProps = {
  color: { hex: '#fff' },
  menuColor: { hex: '#fff' },
  menuBackground: { hex: '#000' },
};

export default Sidebar;
