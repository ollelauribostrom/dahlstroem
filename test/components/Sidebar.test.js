import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from '../../src/components/Sidebar';

const mockData = {
  allDatoCmsCategory: {
    edges: [
      {
        node: {
          name: 'Test',
          slug: '/test',
        },
      },
    ],
  },
};

describe('Tests for Sidebar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Sidebar
        data={mockData}
        color={{ hex: '#000' }}
        menuColor={{ hex: '#000' }}
        menuBackground={{ hex: '#fff' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('uses defaultProps values for colors if none is provided', () => {
    const wrapper = shallow(<Sidebar data={mockData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
