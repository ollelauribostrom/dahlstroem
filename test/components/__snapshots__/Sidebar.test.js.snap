// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Sidebar renders correctly 1`] = `
<mockConstructor
  render={[Function]}
/>
`;

exports[`Tests for Sidebar uses defaultProps values for colors if none is provided 1`] = `
<mockConstructor
  render={[Function]}
/>
`;
